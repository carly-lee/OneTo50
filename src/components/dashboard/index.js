export { default as Dashboard } from './Dashboard';
export { default as Timer } from './Timer';
export { default as NextNumber } from './NextNumber';