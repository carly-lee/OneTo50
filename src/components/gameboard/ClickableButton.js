import React, { PropTypes } from 'react';
import { TouchableOpacity, Text } from 'react-native';

//@flow
export const ClickableButton = ( props )=>{
  const _onPress = ()=>{
    props.onPress();
  };

  return (
    <TouchableOpacity style={ props.style } onPress={ _onPress }>
      { props.child }
    </TouchableOpacity>
  )
};
 
ClickableButton.propTypes = {
  child: PropTypes.element.isRequired,
  onPress: PropTypes.func.isRequired,
}

