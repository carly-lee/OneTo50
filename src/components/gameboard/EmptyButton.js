// @flow
// $FlowIgnore
import { StyleSheet, View } from 'react-native';
import React from 'react';

export const EmptyButton = ()=>(
  <View style={styles.container}></View>
);

const styles = StyleSheet.create({
  container: {
    flex: 0,
    width: 70,
    height: 70,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'white'
  }
});