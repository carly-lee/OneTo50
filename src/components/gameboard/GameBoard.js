//@flow
import React, { PureComponent, PropTypes } from 'react';
import { Text, View, StyleSheet } from 'react-native';

export default class GameBoard extends PureComponent{
  render(){
    return(
      <View style={ styles.container }>
        <Text>GameBoard</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});