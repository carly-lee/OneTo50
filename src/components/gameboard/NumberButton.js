import React, { PureComponent } from 'react';
import { Text, View, StyleSheet } from 'react-native';

//@flow
export default class NumberButton extends PureComponent{
  render(){
    return(
      <View style={ styles.container }>
        <Text>1</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 0,
  },
});