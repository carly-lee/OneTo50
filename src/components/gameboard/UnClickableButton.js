// @flow
// $FlowIgnore
import { View, Text } from 'react-native';
import React, { PropTypes } from 'react';

type Props = {
  child: React$Element<*>,
  style: Object,
};

export const UnClickableButton = ({...props}:Props)=>{
  return (
    <View style={ props.style }>
      { props.child }
    </View>
  );
}
