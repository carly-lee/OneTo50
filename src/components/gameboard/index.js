export { default as GameBoard } from './GameBoard';
export { default as NumberButton } from './NumberButton';