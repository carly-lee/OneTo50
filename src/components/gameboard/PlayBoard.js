import React, { PureComponent } from 'react';
import { View, StyleSheet } from 'react-native';

import { NumberButton } from 'components/gameboard';

//@flow
export default class PlayBoard extends PureComponent{
  allNumbers:Array<number> = [];

  state: {
    buttons:?Array<NumberButton>
  };

  constructor( props ){
    super( props );
    for( let i=0; i<50; i++ ){
      this.allNumbers.push( i+1 );
    }
    this.state = {
      buttons: null,
    };
  }

  componentWillMount(){
    let buttons:Array<NumberButton> = [];
    let n:number;
    for( let i = 0; i<25; i++ ){
      n = this.allNumbers.splice( 0,1 )[0];
      buttons.push( <NumberButton key={ i } number={ n } index={ i } onClickNumber={ this._onClickNumber } clickable={ false } /> );
    }
    this.setState({ buttons });
  }

  _onClickNumber = ( num:number, index:number ):void =>{
    if( this.allNumbers.length > 0 ){
      this._assignNextNumber( index );
    }else{
      console.debug( 'disappear numbers' );
    }
  }

  _assignNextNumber = ( index:number )=>{
    const next:number = this.allNumbers.splice( 0,1 )[0];
    let buttons = [...this.state.buttons];
    const newButton:NumberButton = React.cloneElement( buttons[index], { number: next });
    buttons.splice( index, 1, newButton );
    this.setState({ buttons });
  }

  render(){
    return(
      <View style={ styles.container }>
      { this.state.buttons }
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 0,
    width: 350,
    height: 350,
    flexDirection: 'row',
    flexWrap: 'wrap',
    backgroundColor: 'grey',
  },
});