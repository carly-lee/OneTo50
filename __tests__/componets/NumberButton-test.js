import 'react-native';
import React from 'react';
import { NumberButton } from 'components/gameboard';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe( 'NumberButton', ()=>{
  it( 'renders correctly',()=>{
    const tree = renderer.create( <NumberButton /> );
    expect( tree ).toMatchSnapshot();
  });
});