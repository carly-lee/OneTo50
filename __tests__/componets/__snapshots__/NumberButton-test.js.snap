exports[`NumberButton renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 0,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    1
  </Text>
</View>
`;
